### PR TITLE
Add Spark 3.3.4 Shim

### LINKED ISSUE #9096: [FEA] Add Spark 3.3.4 support
**Is your feature request related to a problem? Please describe.**
Spark 3.3.3 was released, we should add 3.3.4-SNAPSHOT as a new shim.

Note that 3.3.0 was released  June 2022, according to Spark release schedule, its supported for 18 months, so we are getting close to that. It may be a question whether they would do a 3.3.4 release.

### PR DESCRIPTION
**Pull request by razajafri**
This PR adds shims for Spark 3.3.4

**Changes Made:** 
* Ran shimplify `mvn generate-sources -Dshimplify=true -Dshimplify.move=true -Dshimplify.overwrite=true -Dshimplify.add.shim=334 -Dshimplify.add.base=333`
* Fixed the failing test for Sequence length
* Moved some shims around to 334 folder which is the earliest version for those shims

**Tests:** 
All IT tests were run

fixed #9096

### CODE REVIEW

**Comment by razajafri** [comment]
The following tests were failing but I fixed them by pulling in changes from #9962 

This may cause a merge conflict depending on what gets checked in first 

```
==================================================================================================================================================================================================== short test summary info ====================================================================================================================================================================================================
FAILED ../../src/main/python/collection_ops_test.py::test_sequence_too_long_sequence[Integer(not_null)][DATAGEN_SEED=1703023161, INJECT_OOM] - AssertionError: Expected error 'Too long sequence' did not appear in 'py4j.protocol.Py4JJavaError: An error occurred while calling o271142.collectToPython.
FAILED ../../src/main/python/collection_ops_test.py::test_sequence_too_long_sequence[Long(not_null)][DATAGEN_SEED=1703023161] - AssertionError: Expected error 'Too long sequence' did not appear in 'py4j.protocol.Py4JJavaError: An error occurred while calling o271226.collectToPython.
============================================================================================================================================================== 2 failed, 19947 passed, 1732 skipped, 371 xfailed, 303 xpassed, 8532 warnings in 4448.90s (1:14:08) ==============================================================================================================================================================
```

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
build

File: sql-plugin/src/main/spark340/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala [hunk 1]
```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+
```
Comment (jlowe) [inline sql-plugin/src/main/spark340/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala:17]: Whitespace-only change to this file?

File: sql-plugin/src/main/spark340/scala/org/apache/spark/sql/rapids/RapidsCachingReader.scala [hunk 2]
```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+
```
Comment (jlowe) [inline sql-plugin/src/main/spark340/scala/org/apache/spark/sql/rapids/RapidsCachingReader.scala:17]: Whitespace-only change to this file?

File: tests/src/test/spark340/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleTestHelper.scala [hunk 3]
```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+
```
Comment (jlowe) [inline tests/src/test/spark340/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleTestHelper.scala:17]: Whitespace-only change to this file?

**Review by jlowe — APPROVED** [review]
Whitespace nits, lgtm.

**Comment by razajafri** [comment]
Thanks for the review @jlowe I will push a separate PR for the whitespace nits as the CI has passed.

**Comment by jlowe** [comment]
> I will push a separate PR for the whitespace nits as the CI has passed.

The point of the comment was to reduce unnecessary file churn by needlessly touching a file.  Pushing out the fix for that to a separate PR misses the point and _increases_ the unnecessary file churn.  I was fine with it going unfixed as it's pretty minor, not sure it needs to be cleaned up now that it's been done.

**Pull request merged**